server/src/database/migrations/001-initial-migration.ts

```diff
@@ -0,0 +1,384 @@
+          allowNull: false
+        },
+        created_at: DataTypes.DATE,
+        updated_at: DataTypes.DATE
+      }, { transaction });
+      await queryInterface.createTable('course_result', {
+        id: {
+          type: DataTypes.INTEGER,
+          autoIncrement: true,
+          primaryKey: true
+        },
+        user_id: {
+          type: DataTypes.INTEGER,
+          allowNull: false,
+          references: {
+            model: 'user',
+            key: 'id'
+          },
+          onDelete: 'CASCADE',
+          onUpdate: 'CASCADE'
+        },
+        course_instance_id: {
+          type: DataTypes.INTEGER,
+          allowNull: false,
+          references: {
+            model: 'course_instance',
+            key: 'id'
+          },
+          onDelete: 'CASCADE',
+          onUpdate: 'CASCADE'
+        },
+        grade: {
+          type: DataTypes.INTEGER,
+          allowNull: false
+        },
+        credits: {
+          type: DataTypes.INTEGER,
+          allowNull: false
+        },
+        created_at: DataTypes.DATE,
+        updated_at: DataTypes.DATE
+      }, { transaction });
+      await transaction.commit();
+    } catch (error) {
+      await transaction.rollback();
+      console.log(error);
+    }
+  },
+  down: async (queryInterface: QueryInterface): Promise<void> => {
+    const transaction: Transaction = await queryInterface.sequelize.transaction();
+    try {
+      await queryInterface.dropTable('course_result', { transaction });
+      await queryInterface.dropTable('user_assignment_grade', { transaction });
+      await queryInterface.dropTable('course_assignment', { transaction });
+      await queryInterface.dropTable('user_partial_grade', { transaction });
+      await queryInterface.dropTable('course_instance_partial_grade', { transaction });
+      await queryInterface.dropTable('course_translation', { transaction });
+      await queryInterface.dropTable('course_role', { transaction });
+      await queryInterface.dropTable('course_instance', { transaction });
+      await queryInterface.dropTable('course', { transaction });
+      await queryInterface.dropTable('user', { transaction });
+      await queryInterface.dropTable('migrations', { transaction });
+      await queryInterface.sequelize.query('DROP TYPE IF EXISTS enum_course_instance_grading_type;', { transaction });
+      await queryInterface.sequelize.query('DROP TYPE IF EXISTS enum_course_role_role;', { transaction });
+      await queryInterface.sequelize.query('DROP TYPE IF EXISTS enum_course_instance_teaching_method;', { transaction });
+      await queryInterface.sequelize.query('DROP TYPE IF EXISTS enum_course_translation_language;', { transaction });
+      await queryInterface.sequelize.query('DROP TYPE IF EXISTS enum_course_instance_partial_grade_type;', { transaction });
+      await queryInterface.sequelize.query('DROP TYPE IF EXISTS enum_course_instance_partial_grade_platform;', { transaction });
```

Comment:
Is it possible to do this without raw queries? I understood we should avoid them if possible.

Reply:
I could not find a way to do it without raw query. Raw query should not pose any risk here since no parameters are inputted and these are not available to the end user?

queryInterface methods are listed here:
https://sequelize.org/api/v6/class/src/dialects/abstract/query-interface.js~queryinterface

Reply:
I agree, these queries are probably fine. I was just thinking in terms of "best practices" which may be to avoid raw queries whenever possible.